src/qibolab/pulses/pulse.py

```diff
@@ -152,15 +117,31 @@ def __hash__(self):
             )
         )
 
-    def is_equal_ignoring_start(self, item) -> bool:
-        """Check if two pulses are equal ignoring start time."""
-        return (
-            self.duration == item.duration
-            and self.amplitude == item.amplitude
-            and self.frequency == item.frequency
-            and self.relative_phase == item.relative_phase
-            and self.shape == item.shape
-            and self.channel == item.channel
-            and self.type == item.type
-            and self.qubit == item.qubit
-        )
+
+@dataclass
+class Delay:
+    """A wait instruction during which we are not sending any pulses to the
+    QPU."""
+
+    duration: int
+    """Delay duration in ns."""
+    channel: str
+    """Channel on which the delay should be implemented."""
+    type: PulseType = PulseType.DELAY
+    """Type fixed to ``DELAY`` to comply with ``Pulse`` interface."""
+
+
+@dataclass
+class VirtualZ:
+    """Implementation of Z-rotations using virtual phase."""
+
+    duration = 0
+    """Duration of the virtual gate should always be zero."""
+
+    phase: float
+    """Phase that implements the rotation."""
+    channel: Optional[str] = None
+    """Channel on which the virtual phase should be added."""
+    qubit: int = 0
+    """Qubit on the drive of which the virtual phase should be added."""
+    type: PulseType = PulseType.VIRTUALZ
```

Comment:
Even in this case, I'm not user why you need to define the `type`. It's not a `Pulse`, and you don't have to disambiguate wrt that.